lms/static/scripts/frontend_apps/services/vitalsource.js

```diff
@@ -26,6 +26,25 @@ export class VitalSourceService {
     this._authToken = authToken;
   }
 
+  /**
+   * Fetch a book by bookID ("vbid")
+   *
+   * This is currently a fake that waits for a fixed time before returning
+   * hard-coded data.
+   *
+   * @param {string} bookID
+   * @param {number} [fetchDelay] - Dummy delay to simulate slow third-party
+   * @return {Promise<Book>}
+   */
+  async fetchBook(bookID, fetchDelay = 500) {
+    await delay(fetchDelay);
+    const book = bookList.find(b => b.id === bookID);
+    if (!book) {
+      throw new APIError(404, { message: 'Book not found' });
```

Comment:
I'm  not 100% convinced it's appropriate to throw here, but I'm following the pattern of the other methods in this module.

Reply:
An advantage of throwing an error is that there are only two cases for the downstream client to handle: Got the book, didn't get the book (regardless of cause). If the handling was very different downstream for the not-found vs another-error-happened case then that would nullify the advantage.